test/integration/crud/crud.prose.test.ts

```diff
@@ -1079,4 +1080,85 @@ describe('CRUD Prose Spec Tests', () => {
+    // [DRIVERS-2921](https://jira.mongodb.org/browse/DRIVERS-2921)).
+    // Construct a `MongoClient` (referred to as `client`) with
+    // [command monitoring](../../command-logging-and-monitoring/command-logging-and-monitoring.md) enabled to observe
+    // CommandStartedEvents. Perform a `hello` command using `client` and record the `maxBsonObjectSize` and
+    // `maxMessageSizeBytes` values in the response.
+    // Construct a `MongoCollection` (referred to as `coll`) for the collection "db.coll". Drop `coll`.
+    // Use the `create` command to create "db.coll" to workaround [SERVER-95537](https://jira.mongodb.org/browse/SERVER-95537).
+    // Construct the following write model (referred to as `model`):
+    // InsertOne: {
+    //   "namespace": "db.coll",
+    //   "document": { "a": "b".repeat(maxBsonObjectSize - 500) }
+    // }
+    // Construct a list of write models (referred to as `models`) with `model` repeated
+    // `maxMessageSizeBytes / maxBsonObjectSize + 1` times.
+    // Call `client.bulkWrite` with `models`. Pass `BulkWriteOptions` with `ordered` set to `false` and `writeConcern` set to
+    // an unacknowledged write concern. Assert no error occurred. Assert the result indicates the write was unacknowledged.
+    // Assert that two CommandStartedEvents (referred to as `firstEvent` and `secondEvent`) were observed for the `bulkWrite`
+    // command. Assert that the length of `firstEvent.command.ops` is `maxMessageSizeBytes / maxBsonObjectSize`. Assert that
+    // the length of `secondEvent.command.ops` is 1. If the driver exposes `operationId`s in its CommandStartedEvents, assert
+    // that `firstEvent.operationId` is equal to `secondEvent.operationId`. Assert both commands include
+    // `writeConcern: {w: 0}`.
+    // To force completion of the `w:0` writes, execute `coll.countDocuments` and expect the returned count is
+    // `maxMessageSizeBytes / maxBsonObjectSize + 1`. This is intended to avoid incomplete writes interfering with other tests
+    // that may use this collection.
+    let client: MongoClient;
+    let maxBsonObjectSize;
+    let maxMessageSizeBytes;
+    let numModels;
+    const models: AnyClientBulkWriteModel[] = [];
+    const commands: CommandStartedEvent[] = [];
+
+    beforeEach(async function () {
+      const uri = this.configuration.url({
+        useMultipleMongoses: false
+      });
+      client = this.configuration.newClient(uri, { monitorCommands: true });
+      await client.connect();
+      await client.db('db').collection('coll').drop();
+      await client.db('db').createCollection('coll');
+      const hello = await client.db('admin').command({ hello: 1 });
+      maxBsonObjectSize = hello.maxBsonObjectSize;
+      maxMessageSizeBytes = hello.maxMessageSizeBytes;
+      numModels = Math.floor(maxMessageSizeBytes / maxBsonObjectSize) + 1;
+      Array.from({ length: numModels }, () => {
+        models.push({
```

Comment:
A little mixture of approaches here IIUC we're using the Array.from to loop, to push into an array that is scoped to the describe block, I am not seeing the array be reset, granted there's only one test but maybe this should've returned the array to overwrite the variable for each test?

Reply:
Updated to set the array instead of pushing to it.